### PR TITLE
Push/localdevice rsh8

### PR DESCRIPTION
**Pull request by marto83**
To be reviewed after https://github.com/ably/ably-dotnet/pull/887

### CODE REVIEW

File: src/IO.Ably.Shared/Push/LocalDevice.cs [hunk 1]
```diff
@@ -62,10 +62,94 @@ public static LocalDevice Create(string clientId = null, IMobileDevice mobileDev
...
+
+        internal static void ResetDevice(IMobileDevice mobileDevice)
+        {
+            mobileDevice.ClearPreferences(PersistKeys.Device.SharedName);
+            Instance = null;
+        }
+
+        internal static LocalDevice Instance { get; set; }
+
+        internal static LocalDevice GetInstance(IMobileDevice mobileDevice = null, string clientId = null)
```
Comment (tomkirbygreen) [inline src/IO.Ably.Shared/Push/LocalDevice.cs:78]: Why are we defaulting this to `null` if it's never correct to call the API when it *is* `null` as it throws?

**Review by tomkirbygreen — COMMENTED** [review]
Just a question about API design...

File: src/IO.Ably.Shared/Push/LocalDevice.cs (hunk 1 above)
Comment (marto83) [reply src/IO.Ably.Shared/Push/LocalDevice.cs:78]: Good point. I'll update it. I always pass both parameters so I'll remove the nulls

Comment (marto83) [reply src/IO.Ably.Shared/Push/LocalDevice.cs]: I've pushed an update

**Review by tomkirbygreen — APPROVED** [review]

**Pull request merged**